src/AssetRegistry/ATokenSubRegistry.sol

```diff
@@ -0,0 +1,182 @@
+import "../interfaces/ISubRegistry.sol";
+import "../interfaces/IMainRegistry.sol";
+import {FixedPointMathLib} from "../utils/FixedPointMathLib.sol";
+
+/**
+ * @title Sub-registry for Aave Yield Bearing ERC20 tokens
+ * @author Arcadia Finance
+ * @notice The ATokenSubRegistry stores pricing logic and basic information for yield bearing Aave ERC20 tokens for which a direct price feed exists
+ * @dev No end-user should directly interact with the ATokenSubRegistry, only the Main-registry, Oracle-Hub or the contract owner
+ */
+contract ATokenSubRegistry is SubRegistry {
+    using FixedPointMathLib for uint256;
+
+    struct AssetInformation {
+        uint64 assetUnit;
+        address assetAddress;
+        address underlyingAssetAddress;
+        address[] underlyingAssetOracleAddresses;
+    }
+
+    mapping(address => AssetInformation) public assetToInformation;
+
+    /**
+     * @notice A Sub-Registry must always be initialised with the address of the Main-Registry and of the Oracle-Hub
+     * @param mainRegistry The address of the Main-registry
+     * @param oracleHub The address of the Oracle-Hub
+     */
+    constructor(address mainRegistry, address oracleHub)
+        SubRegistry(mainRegistry, oracleHub)
+    {}
+
+    /**
+     * @notice Adds a new asset to the ATokenSubRegistry, or overwrites an existing asset.
+     * @param assetAddress The contract address of the asset
+     * @param assetCreditRatings The List of Credit Ratings for the asset for the different BaseCurrencies.
+     * @dev The list of Credit Ratings should or be as long as the number of baseCurrencies added to the Main Registry,
+     *      or the list must have length 0. If the list has length zero, the credit ratings of the asset for all baseCurrencies is
+     *      is initiated as credit rating with index 0 by default (worst credit rating).
+     * @dev The assets are added/overwritten in the Main-Registry as well.
+     *      By overwriting existing assets, the contract owner can temper with the value of assets already used as collateral
+     *      (for instance by changing the oracleaddres to a fake price feed) and poses a security risk towards protocol users.
+     *      This risk can be mitigated by setting the boolean "assetsUpdatable" in the MainRegistry to false, after which
+     *      assets are no longer updatable.
+     * @dev Assets can't have more than 18 decimals.
+     */
+    function setAssetInformation(
+        address assetAddress,
+        uint256[] calldata assetCreditRatings
+    ) external onlyOwner {
+        address underlyingAddress = IAToken(assetAddress).UNDERLYING_ASSET_ADDRESS();
+        (
+            uint64 assetUnit,
+            address underlyingAssetAddress,
+            address[] memory underlyingAssetOracleAddresses
+            ) = ISubRegistry(IMainRegistry(mainRegistry).assetToSubRegistry(underlyingAddress)).getAssetInformation(underlyingAddress);
+
+        IOraclesHub(oracleHub).checkOracleSequence(
```

Comment:
Check not necessary anymore, since it was already done when underlying asset was added to ERC20subregistry